modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -3931,12 +3946,18 @@ private void _insertIntoLocalizationTable(
 		sb.append(", languageId");
 
 		for (ObjectField objectField : objectFields) {
+			Map<String, Serializable> localizedValues =
+				(Map<String, Serializable>)values.get(
+					objectField.getI18nObjectFieldName());
+
 			if (objectField.isRequired() &&
-				!values.containsKey(objectField.getI18nObjectFieldName()) &&
+				(MapUtil.isEmpty(localizedValues) ||
+				 Validator.isNull(
+					 MapUtil.getString(localizedValues, defaultLanguageId))) &&
 				(workflowAction != WorkflowConstants.ACTION_SAVE_DRAFT)) {
```

Comment:
We should reorder the conditions:
1. (MapUtil.isEmpty(localizedValues) || Validator.isNull(MapUtil.getString(localizedValues, defaultLanguageId))
2. objectField.isRequired()
3. workflowAction != WorkflowConstants.ACTION_SAVE_DRAFT)

Reply:
I'm having hard time to understand that if the FF is off then the code will not check this condition:
`MapUtil.isEmpty(localizedValues) || Validator.isNull(MapUtil.getString(localizedValues, defaultLanguageId))`

If the FF is off, the code cannot require a value for the default language.